File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.cpp
class Solution {
public:
    int specialArray(vector<int>& nums) {
        int N=nums.size();
        vector<int> freq(N+1,0);
        for(auto num:nums){
            freq[min(N,num)]+=1;
        }
        int answer=0;
        for(int i=N;i>0;i--){
            answer+=freq[i];
            if(i==answer){
                return i;
            }
        }
        return -1;
    }
};

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        N=len(nums)
        freq=[0]*(N+1)
        for num in nums:
            freq[min(N,num)]+=1
        number_greater_or_equal=0
        for i in range(N,0,-1):
            number_greater_or_equal+=freq[i]
            if i==number_greater_or_equal:
                return i
        return -1

            

